LogSessionImport/LogSessionImport.cpp

```diff
@@ -0,0 +1,158 @@
+ * exported log session to a new distributed system.
+ *
+ */
+
+#include <iostream>
+#include <PolySyncCore.hpp>
+#include <PolySyncDataModel.hpp>
+#include <PolySyncApplication.hpp>
+#include <PolySyncLogSessionTransfer.hpp>
+
+using namespace std;
+using namespace polysync;
+using namespace datamodel;
+
+/**
+ * @brief SessionImportExample class
+ */
+class SessionImportExample : public DataSubscriber
+{
+
+public:
+
+    SessionImportExample(
+            int sessionId )
+        :
+        _sessionId( sessionId )
+    {
+        // Subscribe to ApplicationEventMessage to determine when
+        // the application connects to the PolySync bus.
+        connectSubscriberMethod< SessionImportExample >(
+                ApplicationEventMessage::getName(),
+                &SessionImportExample::handleEvent,
+                this );
+
+        _application = Application::getInstance();
+
+        _application->attachSubscriber( this );
+    }
+
+private:
+
+    void handleEvent( shared_ptr< Message > message )
+    {
+        if( auto event = getSubclass< ApplicationEventMessage >( message ) )
+        {
+            if( event->getEventKind() == EventKind::Init )
+            {
+                // This is the actual usage of the import API.
+                _importer = new LogSessionImport( _sessionId );
+
+                //
+                // Assign imported nodes to new destination host.
+                //
+
+                // Destination hosts are specified using 'LogSessionTransferTarget'.
+                // The import API requires the transfer target to provide one of the following:
+                // * Interface Address String ("127.0.0.1")
+                // * Manager GUID (562950866709306)
+                // * Host's Id in the SDF
+                LogSessionTransferTarget transferTarget;
+
+                for( const auto & availableNode : _importer->getAvailableNodes() )
+                {
+                    cout << "assigning " << availableNode.getName() << " to 127.0.0.1" << endl;
+
+                    transferTarget.setInterfaceAddress("127.0.0.1");
+
+                    _importer->updateNodeTarget(
+                            transferTarget,
+                            availableNode );
+                }
+
+                // Import can be started with or without a progress callback.
+                // When complete, results will be located at "PSYNC_HOME/rnr_logs/[sessionId]"
+                _importer->start(
+                        this,
+                        &SessionImportExample::handleTransferStatus );
+            }
+        }
+    }
+
+    void handleTransferStatus(
+            const LogSessionTransferStatus & status )
+    {
+        // Status has other information available as well
+        auto state = status.getState();
+
+        cout << "State: ";
+
+        switch( state )
+        {
+            case LogSessionTransferState::Invalid : cout << "Invalid" << endl; break;
```

Comment:
We don't have a clear guide for switch statements yet, but the statements should be on new lines.